src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts

```diff
@@ -1126,9 +1136,20 @@ export class EditDataGridPanel extends GridParentComponent {
 		});
 	}
 
-	onBeforeEditCell(event: Slick.OnBeforeEditCellEventArgs<any>): void {
+	onBeforeEditCell(event: Slick.OnBeforeEditCellEventArgs<any>): boolean {
+		let result = true;
 		this.logService.debug('onBeforeEditCell called with grid: ' + event.grid + ' row: ' + event.row
 			+ ' cell: ' + event.cell + ' item: ' + event.item + ' column: ' + event.column);
+
+		let itemToEdit = event.item[event.cell].displayValue;
+
+		// allow-any-unicode-next-line
+		if (Services.DBCellValue.isDBCellValue(itemToEdit) && (itemToEdit.displayValue.indexOf('\u0000') !== -1 || itemToEdit.displayValue.indexOf('↵') !== -1 || itemToEdit.displayValue.indexOf('⏎') !== -1)) {
```

Comment:
we only need to block the editing if the value contains NULL and line break at the same time.